src/Libraries/RevitNodes/Elements/Parameter.cs

```diff
@@ -40,5 +205,190 @@ public override string ToString()
+
+            // get document and open transaction
+            Document document = Application.Document.Current.InternalDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            try
+            {
+                // get current shared parameter file
+                string sharedParameterFile = document.Application.SharedParametersFilename;
+
+                // if the file does not exist, throw an error
+                if (sharedParameterFile == null || !System.IO.File.Exists(sharedParameterFile))
+                    throw new System.Exception(Properties.Resources.NoSharedParameterFileFound);
+
+                // Apply selected parameter categories
+                CategorySet categories = ToCategorySet(categoryList);
+
+                // Create new parameter group if it does not exist yet
+                DefinitionGroup groupDef = document.Application.OpenSharedParameterFile().Groups.get_Item(groupName);
+                if (groupDef == null)
+                {
+                    groupDef = document.Application.OpenSharedParameterFile().Groups.Create(groupName);
+                }
+
+                // If the parameter definition does not exist yet, create it
+                if (groupDef.Definitions.get_Item(parameterName) == null)
+                {
+                    ExternalDefinition def = groupDef.Definitions.Create(new ExternalDefinitionCreationOptions(parameterName, parameterType)) as ExternalDefinition;
+
+                    // Apply instance or type binding
+                    Binding bin = (instance) ? (Binding)document.Application.Create.NewInstanceBinding(categories) : (Binding)document.Application.Create.NewTypeBinding(categories);
+                    document.ParameterBindings.Insert(def, bin, parameterGroup);
+                }
+
+            }
+            catch (System.Exception e)
+            {
+                TransactionManager.Instance.ForceCloseTransaction();
+                throw e;
+            }
+
+            TransactionManager.Instance.TransactionTaskDone();
+        }
+
+
+        #endregion
+
+        #region Project Parameters
+
+        /// <summary>
+        /// Create Project Parameter
+        /// </summary>
+        /// <param name="parameterName">Name</param>
+        /// <param name="groupName">Group of the parameter for shared parameters</param>
+        /// <param name="type">Parameter Type</param>
+        /// <param name="group">Parameter Group</param>
+        /// <param name="instance">Is instance parameter, otherwise its a type parameter</param>
+        /// <param name="categoryList">List of categories this parameter applies to</param>
+        public static void CreateProjectParameter(string parameterName, string groupName, string type, string group, bool instance, [DefaultArgumentAttribute("null")]System.Collections.Generic.IEnumerable<Category> categoryList)
```

Comment:
can you split this into multiple lines- 

additionally can you make an AllCategories token that we can use instead of null? the intention will become 100% more clear.